InnerEye/ML/SSL/datamodules_and_datasets/cxr_datasets.py

```diff
@@ -0,0 +1,155 @@
+    This class is just a shorthand notation for this double inheritance.
+    """
+    pass
+
+
+class RSNAKaggleCXR(InnerEyeCXRDatasetWithReturnIndex):
+    """
+    Dataset class to load the RSNA Chest-Xray training dataset. Use all the data for train and val. No test data
+    implemented.
+    """
+
+    def _prepare_dataset(self) -> None:
+        if self.train:
+            self.dataset_dataframe = pd.read_csv(self.root / "dataset.csv")
+            self.targets = self.dataset_dataframe.label.values.astype(np.int64)
+            self.subject_ids = self.dataset_dataframe.subject.values
+            self.indices = np.arange(len(self.dataset_dataframe))
+            self.filenames = [self.root / f"{subject_id}.dcm" for subject_id in self.subject_ids]
+        else:
+            # No test set implemented for this data class.
+            self.indices = []
+            self.filenames = []
+
+    @property
+    def num_classes(self) -> int:
+        return 2
+
+
+class NIH(InnerEyeCXRDatasetWithReturnIndex):
+    """
+    Dataset class to load the NIH Chest-Xray dataset. Use the full data for training and validation (including
+    the official test set by default).
+    """
+
+    def __init__(self,
+                 root: str,
+                 use_full_dataset_for_train_and_val: bool = True,
+                 **kwargs: Any) -> None:
+        self.use_full_dataset_for_train_and_val = use_full_dataset_for_train_and_val
+        super().__init__(root=root, **kwargs)
+
+    def _prepare_dataset(self) -> None:
+        self.dataset_dataframe = pd.read_csv(self.root / "Data_Entry_2017.csv")
+        # To use full dataset (incl. official test set for train & val of SSL models, no test set)
+        if self.use_full_dataset_for_train_and_val:
+            self.subject_ids = self.dataset_dataframe["Image Index"].values if self.train else []
+        # To exclude official test set from train & val
+        else:
+            train_ids = pd.read_csv(self.root / "train_val_list.txt", header=None).values.reshape(-1)
+            is_train_val_ids = self.dataset_dataframe["Image Index"].isin(train_ids).values
+            self.subject_ids = np.where(is_train_val_ids)[0] if self.train else np.where(~is_train_val_ids)[0]
+        self.indices = np.arange(len(self.subject_ids))
+        self.filenames = [self.root / f"{subject_id}" for subject_id in self.subject_ids]
+
+
+class CheXpert(InnerEyeCXRDatasetWithReturnIndex):
+    """
+    Dataset class to load the CheXpert dataset.
+    """
+
+    def _prepare_dataset(self) -> None:
+        if self.train:  # for train AND val
+            self.dataset_dataframe = pd.read_csv(self.root / "train.csv")
+        else:  # test set (unused in SSL training)
+            self.dataset_dataframe = pd.read_csv(self.root / "valid.csv")
+        # Remove lateral shots
+        self.dataset_dataframe = self.dataset_dataframe.loc[self.dataset_dataframe["Frontal/Lateral"] == "Frontal"]
```

Comment:
maybe we can flag this feature in case someone wants to use both frontal and lateral scans later.

Reply:
True, added a flag